### PR TITLE
[Intel MKL] Upgrading giflib to fix CVE-2019-15133

### PR DESCRIPTION
**Pull request by claynerobison**
# [CVE-2019-15133](https://nvd.nist.gov/vuln/detail/CVE-2019-15133)

**NVD**: 2019/08/17 - CVSS v2.0 Base Score: 4.3 - CVSS v3.0 Base Score: 6.5
In GIFLIB before 2019-02-16, a malformed GIF file triggers a divide-by-zero exception in the decoder function DGifSlurp in dgif_lib.c if the height field of the ImageSize data structure is equal to zero.

## **References to Advisories, Solutions, and Tools**

Source | Link | Type
---- | ---- | ----
MISC | [bugs.chromium.org](https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=13008) | Mailing List, Third Party Advisory
UBUNTU | [usn.ubuntu.com](https://usn.ubuntu.com/4107-1/) | Third Party Advisory

### CODE REVIEW

**Comment by claynerobison** [comment]
@rthadur This CVE doesn't just apply to MKL builds.

File: tensorflow/workspace.bzl [hunk 1]
```diff
@@ -287,12 +287,12 @@ def tf_repositories(path_prefix = "", tf_repo_name = ""):
...
         build_file = clean_dep("//third_party:gif.BUILD"),
-        sha256 = "34a7377ba834397db019e8eb122e551a49c98f49df75ec3fcc92b9a794a4f6d1",
-        strip_prefix = "giflib-5.1.4",
+        sha256 = "31da5562f44c5f15d63340a09a4fd62b48c45620cd302f77a6d9acf0077879bd",
+        strip_prefix = "giflib-5.2.1",
         system_build_file = clean_dep("//third_party/systemlibs:gif.BUILD"),
         urls = [
-            "https://storage.googleapis.com/mirror.tensorflow.org/ufpr.dl.sourceforge.net/project/giflib/giflib-5.1.4.tar.gz",
-            "http://pilotfiber.dl.sourceforge.net/project/giflib/giflib-5.1.4.tar.gz",
+            "https://storage.googleapis.com/mirror.tensorflow.org/ufpr.dl.sourceforge.net/project/giflib/giflib-5.2.1.tar.gz",
```
Comment (penpornk) [inline tensorflow/workspace.bzl]:
```suggestion
            "https://storage.googleapis.com/mirror.tensorflow.org/pilotfiber.dl.sourceforge.net/project/giflib/giflib-5.2.1.tar.gz",
```

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you very much for the PR!

**Review by penpornk — APPROVED** [review]
Thank you for the quick fix!

**Comment by claynerobison** [comment]
@penpornk @gunan May I suggest that this be cherry-picked?

**Comment by gunan** [comment]
@goldiegadde this should be cherrypicked into 1.15 and 2.0

**Comment by penpornk** [comment]
@gunan @claynerobison @goldiegadde I see a Windows Bazel failure (in some other PR's test results) that seems to originated from this PR.

```
gif.lib(gif_font.obj) : error LNK2019: unresolved external symbol strtok_r referenced in function GifDrawBoxedText8x8
```
[Full log](https://source.cloud.google.com/results/invocations/bd42ab39-0f56-4edc-b789-d16d76348800/log).

(The error doesn't show up in this PR's test results because the Windows tests were failing from CSRSparseMatrix changes at the time.)

Please don't cherrypick this PR yet. I'll fix this and cherrypick the changes afterwards.

**Comment by penpornk** [comment]
Since this is affecting all other Windows test, we are reverting the PR for now.

**Comment by mihaimaruseac** [comment]
Please assign to me (or notify me) when fixing

**Pull request merged**